src/zcash/prf.h

```diff
@@ -0,0 +1,24 @@
+/*
+Zcash uses SHA256Compress as a PRF for various components
+within the zkSNARK circuit.
+*/
+
+#ifndef _PRF_H_
+#define _PRF_H_
+
+#include "uint256.h"
+
+uint256 PRF(bool a, bool b, bool c, bool d,
+            uint256 x,
+            uint256 y);
+
+template<unsigned char T>
+uint256 PRF_addr(uint256 a_sk)
+{
+    uint256 y;
+    *(y.begin()) = T;
```

Comment:
This is byte-order-dependent. ``uint256 y = uint256(T);`` would be byte-order-independent. (I think that's correct.)